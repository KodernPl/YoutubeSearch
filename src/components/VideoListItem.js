import React from 'react'

const VideoListItem = ({video, onVideoSelect}) => {
    return (
        <div>
            <li onClick={() => onVideoSelect(video)} className="list-group-item">
                <div className="video-list media">
                    <div className="media-left">
                        <img className="media-object" src={video.snippet.thumbnails.default.url}/>
                     </div>

                     <div>
                     <div className="media-heading">{video.snippet.title}</div>
                     </div>
                </div>
            </li>
        </div>
    )
}

export default VideoListItem