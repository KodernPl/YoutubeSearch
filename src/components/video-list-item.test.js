import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import VideoListItem from './video-list-item';

const onVideoSelectMock = jest.fn();

beforeEach(() => {
  onVideoSelectMock.mockReset();
});

const mockVideo = {
  snippet: {
    thumbnails: {
      medium: {
        url: 'TEST',
      },
    },
  },
};

const renderComponent = () => {
  return shallow((
    <VideoListItem
      video={mockVideo}
      onVideoSelect={onVideoSelectMock}
    />
  ));
};

test('renders correctly', () => {
  const component = renderComponent();

  expect(toJson(component)).toMatchSnapshot();
});

test('should call onVideoSelect if clicked', () => {
  const component = renderComponent();
  component.simulate('click', {});

  expect(onVideoSelectMock).toBeCalledWith(mockVideo);
});
