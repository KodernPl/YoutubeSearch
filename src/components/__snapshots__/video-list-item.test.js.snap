// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<video-list-item__VideoListItemWrapper
  onClick={[Function]}
>
  <video-list-item__Video>
    <img
      alt="TEST"
      src="TEST"
    />
  </video-list-item__Video>
  <video-list-item__Paragraph />
</video-list-item__VideoListItemWrapper>
`;
